Develop1

SashiFazya opened the pull request.

SashiFazya committed:
Написаны тесты без моков

SashiFazya committed:
Добавлены моки
Добавлен отчет по покрытию jacoco

nikolay-977 commented on src/test/java/TestBurger.java:
```diff
@@ -0,0 +1,70 @@
+    Ingredient mockIngredient;
+    @Mock
+    Bun mockBun;
+    Burger burger = new Burger();
```
Необходимо исправить: Для инстанс-переменных следует указывать модификатор доступа. Необходимо исправить по всему проекту

nikolay-977 commented on src/test/java/TestBurger.java:
```diff
@@ -0,0 +1,70 @@
+    Burger burger = new Burger();
+    Bun bun = new Bun("с кунжутом", 3.25F);
+    Bun bun2 = new Bun("с отрубями", 2.75F);
+    Ingredient sauce = new Ingredient(IngredientType.SAUCE, "кетченез", 1.5F);
```
Необходимо исправить: Тесты для класса Burger должны быть изолированы от классов Bun и Ingredient. При написании тестов для класса Burger во всех кейсах используй моки и стабы классов Bun и Ingredient, а не реальные объекты.

nikolay-977 commented on src/test/java/TestBurger.java:
```diff
@@ -0,0 +1,70 @@
+        burger.setBuns(bun);
+        burger.addIngredient(sauce);
+        burger.addIngredient(filling1);
+        System.out.println(burger.getReceipt());
```
Можно улучшить: если System.out.println() используется для отладки, лучше использовать логирование. https://www.baeldung.com/java-logging-intro

nikolay-977 commented on src/test/java/TestIngredient.java:
```diff
@@ -0,0 +1,38 @@
+        float ingredPrice = (float) 13.0987;
+        Ingredient ingredient = new Ingredient(ingredType, ingredName, ingredPrice);
+        System.out.println(ingredType);
+        assertEquals("тип проверь", ingredient.getType(), ingredType);
```
Необходимо исправить порядок аргументов в методе assertEquals. В методе assertEquals порядок аргументов должен быть следующий: assertEquals(expected, actual) либо assertEquals(message, expected, actual). Иначе в случае фейла проверки, сообщение об ошибке может ввести в заблуждение, что увеличит время локализации дефекта.

nikolay-977 commented on src/test/java/TestIngredient.java:
```diff
@@ -0,0 +1,38 @@
+    }
+
+    @Test
+    public void test1() {
```
Необходимо исправить: Наименование тестового метода должно отражать суть проверки

SashiFazya committed:
TestBurger: все тесты изолировала моками, исправила порядок аргументов в assert, добавила модификаторы доступа к инстанс-переменным,удалила позорище с sout. Остались тесты с несколькими assert, но считаю делить их не очень правильно, а проверка только размеров массива ингредиентов недостаточна.
TestBun: исправила порядок аргументов в assert.
TestIngredient: исправила порядок аргументов в assert, разделила тест на три отдельных, корректно именовала тесты, добавила один общий пресет для всех трех тестов.
jacoco перегенерила

SashiFazya committed:
Везде поставила модификаторы доступа для переменных
TestBun: вынесла общий пресет в Before
jacoco перегенерила
